File: Basket/src/com/lyonsys/basket/Product.java
package com.lyonsys.basket;

public class Product {


	private final String productId;
	private final String description;

	public Product(String productId, String description) {
		this.productId = productId;
		this.description = description;
	}

	public String getProductId() {
		return productId;
	}

	public String getDescription() {
		return description;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result
				+ ((productId == null) ? 0 : productId.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Product other = (Product) obj;
		if (productId == null) {
			if (other.productId != null)
				return false;
		} else if (!productId.equals(other.productId))
			return false;
		return true;
	}
	
	
	
}

File: scala/Basket/src/com/lyonsys/basket/Product.scala
package com.lyonsys.basket

class Product (_name: String, _desciption:String) extends Equals {
  
  def name = _name;
  
  def canEqual(other: Any) = {
    other.isInstanceOf[com.lyonsys.basket.Product]
  }

  override def equals(other: Any) = {
    other match {
      case that: com.lyonsys.basket.Product => that.canEqual(this) && that.name == this.name;
      case _ => false
    }
  }

  override def hashCode():Int = {
     31  + (if (name == null) 0  else name.hashCode);
  }
}